### PR TITLE
[MM-29921] fix custom sound not playing when receiving a notification

### LINKED ISSUE #1395: 4.6.0 does not play notification sound on Windows OS
I confirm (by marking "x" in the [ ] below: [x]):

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [x] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**
4.6.0 does not play notification sound on Windows OS

**Environment**
- Operating System: Windows 7, Windows 10 1909, 2004, 20H2
- Mattermost Desktop App version: 4.6.0 (exe installer)
- Mattermost Server version: 5.28.0 

**Steps to reproduce**
receive notification in desktop app
**Expected behavior**
when receiving notification, see toast (banner) notification and hear a notification sound
**Observed behavior**
when receiving notification, see toast (banner) notification, but no notification sound is played

**Notes**
To test, I spun up fresh installs of Windows 7, Windows 10 1909, Windows 10 2004 & Windows 10 20H2.  Also fresh installs of Solus Budgie, Linux Mint 20.  On the windows installs, I used the .exe installer.  Of these, only the Linux desktop app played the notification sounds.  On each virtual machine, I was able to select the new custom notification sound and hear them when selecting one, but no sound ever played when receiving notifications.  If I install 4.5.4 on windows, notification sounds played as expected.

Anyone else notice this behavior on windows or am I missing something?

### PR DESCRIPTION
**Pull request by Willyfrog**
**Summary**

Fix notifications not playing custom sounds

**Issue link**
fixes #1395 
[Jira Ticket](https://mattermost.atlassian.net/browse/MM-29921)

**Test Cases**
- ensure the webapp is not loaded with the same user in a browser tab
- go to account settings, notifications and choose a custom sound
- from another user, send an at-mention

Expected: custom sound is played

### CODE REVIEW

**Comment by amyblais** [comment]
@Willyfrog Do you know if there was a recent change that caused this?

**Comment by Willyfrog** [comment]
> @Willyfrog Do you know if there was a recent change that caused this?

Yes, my last minute change to a crash did this. It was apparently safe and if you had a browser open in the background you wouldn't notice that sound was coming from the tab and not the app😥

**Comment by amyblais** [comment]
@nevyangelova @devinbinnie FYI this is for v4.6.1 dot release.

File: src/browser/js/notification.js [hunk 1]
```diff
@@ -46,7 +46,9 @@ export default class EnhancedNotification extends OriginalNotification {
     }
 
     const isWin7 = (process.platform === 'win32' && osVersion.isLowerThanOrEqualWindows8_1() && DEFAULT_WIN7);
-    const customSound = !options.silent && ((options.data && options.data.soundName && options.data.soundName !== 'None') || isWin7);
+    const customSound = !options.silent && ((options.data && options.data.soundName !== 'None' && options.data.soundName) || isWin7);
+    console.log(`customSound: ${customSound}`);
```
Comment (devinbinnie) [inline src/browser/js/notification.js:50]: Should probably remove these log statements?

**Review by devinbinnie — APPROVED** [review]

**Review by nevyangelova — APPROVED** [review]
Oops haha LGTM!

**Review by jgilliam17 — APPROVED** [review]
Thank you @Willyfrog 
Tested, looks good to merge.
- Verified custom sounds play when receiving notification on the desktop app.

**Pull request merged**